src/plotting/plot.component.test.tsx

```diff
@@ -1,267 +1,68 @@
-      expect(mockVictoryScatter.mock.calls.length).toEqual(
-        testPlotDatasets.length
-      );
-      expect(mockVictoryLine).not.toHaveBeenCalled();
-
-      for (let i = 0; i < mockVictoryScatter.mock.calls.length; i++) {
-        expect(mockVictoryScatter.mock.calls[i][0]).toEqual(
-          expect.objectContaining({
-            data: testPlotDatasets[i].data,
-            x: 'test x-axis',
-            y: testPlotDatasets[i].name,
-          })
-        );
-      }
-    });
-
-    it('renders a line plot with the correct elements passed the correct props', () => {
-      props = {
-        ...props,
-        title: 'line plot',
-        type: 'line',
-        XAxisSettings: { scale: 'linear' },
-        YAxesSettings: { scale: 'log' },
-      };
-
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const { default: Plot } = require('./plot.component');
-
-      render(<Plot {...props} />);
-
-      expect(mockVictoryChart).toHaveBeenCalledWith(
-        expect.objectContaining({
-          scale: { x: 'linear', y: 'log' },
-        })
-      );
-      expect(mockVictoryLabel).toHaveBeenCalledWith(
-        expect.objectContaining({
-          text: 'line plot',
-        })
-      );
-      expect(mockVictoryLegend).toHaveBeenCalledWith(
-        expect.objectContaining({
-          data: testPlotDatasets.map((dataset) => {
-            return { name: dataset.name, symbol: { fill: '#e31a1c' } };
-          }),
-        })
-      );
-      expect(mockVictoryGroup).toHaveBeenCalled();
-      expect(mockVictoryScatter.mock.calls.length).toEqual(
-        testPlotDatasets.length
-      );
-      expect(mockVictoryLine.mock.calls.length).toEqual(
-        testPlotDatasets.length
-      );
-
-      for (let i = 0; i < mockVictoryScatter.mock.calls.length; i++) {
-        expect(mockVictoryScatter.mock.calls[i][0]).toEqual(
-          expect.objectContaining({
-            data: testPlotDatasets[i].data,
-            x: 'test x-axis',
-            y: testPlotDatasets[i].name,
-          })
-        );
-        expect(mockVictoryLine.mock.calls[i][0]).toEqual(
-          expect.objectContaining({
-            data: testPlotDatasets[i].data,
-            x: 'test x-axis',
-            y: testPlotDatasets[i].name,
-          })
-        );
-      }
-    });
-
-    it('renders no plot if the user has not selected any channels to plot', () => {
-      props.selectedChannels = [];
+    expect(view.asFragment()).toMatchSnapshot();
+  });
 
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const { default: Plot } = require('./plot.component');
+  it('updates options object correctly', () => {
+    const { rerender, asFragment } = render(<Plot {...props} />);
 
-      render(<Plot {...props} />);
+    props = {
+      ...props,
+      title: 'line plot',
+      type: 'line',
+      XAxisSettings: { scale: 'linear' },
+      YAxesSettings: { scale: 'log' },
+    };
```

Comment:
Can we also check that `props.XAxis` updates on the graph as well?

Reply:
Thanks for catching that 👍